### DISCUSSION TITLE
Facing issue while setup browsershot in server

### DISCUSSION THREAD

**Discussion by shabnam227** [spatie/browsershot#828] [General]
The command "PATH=$PATH:/home/ubuntu/.nvm/versions/node/v20.11.1/bin NODE_PATH=`npm root -g` node '/var/www/html/order_dev/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/894557859-0799827001710739477\/index.html","action":"screenshot","options":{"type":"png","path":"orderbill\/new.png","args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"ignoreHttpsErrors":true}}'" failed. Exit Code: 1(General error) Working directory: /var/www/html/order_dev/public Output: ================ Error Output: ================ /usr/bin/env: 'node': No such file or directory Error: Could not find Chrome (ver. 122.0.6261.128). This can occur if either 1. you did not perform an installation before running the script (e.g. `npx puppeteer browsers install chrome`) or 2. your cache path is incorrectly configured (which is: /var/www/.cache/puppeteer). For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration. at ChromeLauncher.resolveExecutablePath (/var/www/html/order_dev/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:285:27) at ChromeLauncher.executablePath (/var/www/html/order_dev/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:196:25) at ChromeLauncher.computeLaunchArguments (/var/www/html/order_dev/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:89:37) at async ChromeLauncher.launch (/var/www/html/order_dev/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:70:28) at async callChrome (/var/www/html/order_dev/vendor/spatie/browsershot/bin/browser.cjs:92:23)

i setup v20.11.1
aslov
 whereis node
node: /usr/local/bin/node /home/ubuntu/.nvm/versions/node/v20.11.1/bin/node   this is the path that i gavi in my code 
Route::get('test-html', function () {
    $html = View::make('bill_template')->render();
    Browsershot::html($html)

     ->setIncludePath('$PATH:/home/ubuntu/.nvm/versions/node/v20.11.1/bin')
  
    ->save('orderbill/new.png');  this is working fine in my local system but when i do same thing on serve i am facing issue 

   

});

**Comment by NikolaCava** [comment]
For me it the problem was that I installed chrome in an inaccessible directory. I added a config file "project-directory/.puppeteerrc.cjs", with this:

```
const {join} = require('path');

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
  // Changes the cache location for Puppeteer.
  cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
};
```

Chek this out https://pptr.dev/guides/configuration#configuration-files. This did fix the error, then I got some errors for missing dependencies, I installed them. But now I get 

> Error: Failed to launch the browser process! chrome_crashpad_handler: --database is required Try 'chrome_crashpad_handler --help' for more information. [623245:623245:0515/102917.466135:ERROR:socket.cc(120)] recvmsg: Connection reset by peer (104) TROUBLESHOOTING: https://pptr.dev/troubleshooting at Interface.onClose (/var/www/project-name/node_modules/@puppeteer/browsers/lib/cjs/launch.js:310:24) at Interface.emit (node:events:531:35) at Interface.close (node:internal/readline/interface:528:10) at Socket.onend (node:internal/readline/interface:254:10) at Socket.emit (node:events:531:35) at endReadableNT (node:internal/streams/readable:1696:12) at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
